superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx

```diff
@@ -61,7 +61,7 @@ export type ExploreQuery = QueryResponse & {
 };
 
 export interface ISimpleColumn {
-  name?: string | null;
+  column_name?: string | null;
```

Comment:
since we are condensing `Query` to use the `TableColumn` model object we need make the names match
https://github.com/apache/superset/blob/f2b512f0dc8514509953e67fb70faacf92868a92/superset/connectors/base/models.py#L602

Before we were managing our on dict inside extra, and I made is `name` originally